contracts/game.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Game is Ownable {
+  IERC20 private _token;
+  address [] judges;
+  address payable[] playersSeekingVerification;
+  address payable[] playersVerifiedToWin;
+  address payable[] winners;
+
+  event PlayerSeeksVerification(address indexed _player);
+  event PlayerVerifiedToWin(address indexed _player, address indexed _judge);
+  event PlayerWon(address indexed player);
+
+  constructor(IERC20 token) {
+    _token = token;
+  }
+
+  modifier judgeOnly {
+    require(isJudge(msg.sender), "Must be a judge");
+    _;
+  }
+
+  function isJudge(address judge) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < judges.length; i++) {
+      if (judge == judges[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 100 * 10**18;
+  }
+
+  function hasTooManyTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 1000 * 10**18;
+  }
+
+  function hasBeenVerifiedToWin(address player) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < playersVerifiedToWin.length; i++) {
+      if (player == playersVerifiedToWin[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasNotWonBefore(address player) view public returns (bool) {
+    return !hasWon(player);
+  }
+
+  function seekVerification() public {
+    require(hasTokens(msg.sender), "Need 100 FWEB3 tokens");
+    require(!hasTooManyTokens(msg.sender), "Need less than 1,000 FWEB3 tokens");
+    playersSeekingVerification.push(payable(msg.sender));
+    emit PlayerSeeksVerification(msg.sender);
+  }
+
+  function win() public {
+    require(hasTokens(msg.sender), "Need 100 FWEB3 tokens");
+    require(!hasTooManyTokens(msg.sender), "Need less than 1,000 FWEB3 tokens");
```

Comment:
Why do we want to stop people who have too many tokens from winning? Maybe instead of making this a requirements, you can still be a winner but you just won’t get a token distribution (I can always just send my tokens to another account if I want to cheat). So either move the token distribution to another function or make it so that this send is nested in an if statement.

Reply:
The reason is for people who've already won by getting tokens. But you're right, people should still be able to "win". Will rethink this.